src/caf/toolkit/cost_utils.py

```diff
@@ -79,6 +83,11 @@ def check_df_col_names(self) -> CostDistribution:
             if col_val not in self.df:
                 err_cols.update({col_name: col_val})
 
+        # Add in the weighted_avg_col if not already in df
+        cols.update({"weighted_avg_col": getattr(self, "weighted_avg_col")})
+        if self.weighted_avg_col not in self.df.columns:
+            self.df[self.weighted_avg_col] = self.df[self.avg_col]
```

Comment:
How easy would it be to add in a quick test to cover this case?

As this is expected default behaviour when the weighted col is missing, I'd like a test to flag if this ever changes in future